mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java

```diff
@@ -0,0 +1,458 @@
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.AttributeMarshaller;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.ParameterCorrector;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.StringListAttributeDefinition;
+import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraintDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.client.helpers.MeasurementUnit;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.operations.validation.EnumValidator;
+import org.jboss.as.controller.operations.validation.IntRangeValidator;
+import org.jboss.as.controller.operations.validation.ParameterValidator;
+import org.jboss.as.controller.operations.validation.StringLengthValidator;
+import org.jboss.as.controller.registry.AliasEntry;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.controller.transform.PathAddressTransformer;
+import org.jboss.as.controller.transform.TransformationContext;
+import org.jboss.as.controller.transform.description.AttributeConverter;
+import org.jboss.as.controller.transform.description.DiscardAttributeChecker;
+import org.jboss.as.controller.transform.description.DiscardPolicy;
+import org.jboss.as.controller.transform.description.DynamicDiscardPolicy;
+import org.jboss.as.controller.transform.description.RejectAttributeChecker;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.modcluster.ModClusterServiceMBean;
+import org.jboss.modcluster.config.impl.SessionDrainingStrategyEnum;
+
+/**
+ * {@link org.jboss.as.controller.ResourceDefinition} implementation for the core mod_cluster configuration resource.
+ *
+ * @author Radoslav Husar
+ */
+public class ProxyConfigurationResourceDefinition extends ChildResourceDefinition<ManagementResourceRegistration> {
+
+    private static final String UNDERTOW_LISTENER_CAPABILITY_NAME = "org.wildfly.undertow.listener";
+
+    static final PathElement LEGACY_PATH = PathElement.pathElement("mod-cluster-config", "configuration");
+    static final PathElement WILDCARD_PATH = pathElement(PathElement.WILDCARD_VALUE);
+
+    static PathElement pathElement(String name) {
+        return PathElement.pathElement("proxy", name);
+    }
+
+    public enum Capability implements org.jboss.as.clustering.controller.Capability {
+        SERVICE("org.wildfly.mod_cluster.service", ModClusterServiceMBean.class),
```

Comment:
I'm wondering about the compatibility of the change here. The capability is public and was previously not dynamic. It's now dynamic which seems like a breaking change since it's a public capability. @jmesnil WDTY?

Reply:
Are you confusing this with https://github.com/wildfly/wildfly/blob/master/mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterConfigResourceDefinition.java#L72 ?

Reply:
I'm not sure TBH. I noticed on there is a capability change PR https://github.com/wildfly/wildfly-capabilities/pull/25/files#diff-671c935d38723b5c603f9e04708f4846L10 which changes it. And this change set removed and constant of it https://github.com/wildfly/wildfly/pull/11317/files#diff-8fdcb0b77f2875d1184e1dfa324f7abbL62. Maybe it's not an issue though.

Reply:
@jamezp Actually, the PR to capabilities was out of date -- updated now. No changes in this PR though.